en/02_Developer_Guides/00_Model/02_Relations.md

```diff
@@ -570,6 +570,100 @@ the best way to think about it is that the object where the relationship will be
 `Product` => `Category`, the `Product` model should contain the `many_many` side of the relationship, because it is much
 more likely that the user will select categories for a product than vice-versa.
 
+## Eager loading
+
+Querying nested relationships inside a loop using the ORM is prone to the N + 1 query problem. To illustrate the N + 1 query problem, imagine a scenario where there are Teams with many child Players
+
+```php
+class Team extends DataObject
+{
+    private static $has_many = [
+        'Players' => Player::class,
+    ];
+}
+```
+
+To retrieve teams and their players:
+
+```php
+$teams = Team::get();
+
+foreach ($teams as $team) {
+    foreach ($team->Players() as $player) {
+        echo $player->FirstName;
+    }
+}
+```
+
+In this case the loop will execute one query to retrieve all the teams and then an additional query for each team to retrieve its players. If we have 20 teams this loop would run 21 queries - one to get all the teams and then 20 more queries to get the players for each team.
+
+```sql
+# Retrieve all the teams
+# Note this is not the exact SQL that would generated it is just for demonstration
+SELECT * FROM Team;
+
+# Retrieve the players for all the teams in 20 separate queries:
+SELECT * FROM Player WHERE TeamID = 1
+SELECT * FROM Player WHERE TeamID = 2
+SELECT * FROM Player WHERE TeamID = 3
+SELECT * FROM Player WHERE TeamID = ...
+```
+
+The N + 1 query problem can be alleviated using eager loading which in this example will reduce this down to just two queries. We do this by passing the relationships that should be eagerly loaded to the `eagerLoad()` method on [DataList](api:SilverStripe\ORM\DataList):
+
+```php
+$teams = Team::get()->eagerLoad('Players')
+```
+
+With eager loading now only two queries will be executed:
+
+```sql
+# Retrieve all the teams
+SELECT * FROM Team
+
+# Retrieve all the players for the teams in a single query:
+SELECT * FROM Player WHERE TeamID IN (1, 2, 3, ...)
+```
+
+Eager load multiple relationships by passing additional arguments separated by a comma to the `eagerLoad()` method. In this example `Players` and `Fans` are `has_many` relationships on the `Team` model:
```

Comment:
```suggestion
Suppose we have the following architecture and relationships between classes:
```php
class Team extends DataObject
{
    private static $has_many = [
        'Players' => Player::class,
        'Fans' => Fan::class,
    ];
}

class Player extends DataObject
{
    private static $has_one = [
        'Team' => Team::class,
    ];
    
    private static $many_many = [
        'Games' => Game::class,
    ];
}

class Game extends DataObject
{
    private static $has_many = [
         'Players' => Player::class,
        'Officials' => Official::class,
        'Sponsors' => Sponsor::class,
    ];
}
In this example, to get all Players and Fans, you can use eager loading multiple relationships by passing additional arguments separated by a comma to the `eagerLoad()` method:
```

Reply:
Updated